libs/core/src/lib/carousel/carousel.component.ts

```diff
@@ -0,0 +1,404 @@
+        if (!this.loop && this.currentActiveSlidesStartIndex <= 0) {
+            return;
+        } else {
+            this.previous();
+        }
+    }
+
+    /** Transitions to the previous slide in the carousel. */
+    public previous(): void {
+        this.rightButtonDisabled = false;
+        this._adjustActiveItemPosition(SlideDirection.PREVIOUS);
+        this._carouselService.pickPrevious(this.dir);
+        this._notifySlideChange(SlideDirection.PREVIOUS);
+        this._changeDetectorRef.markForCheck();
+    }
+
+    /** Transitions to the next slide in the carousel. */
+    public next(): void {
+        // Moving to next slide
+        this.leftButtonDisabled = false;
+        this._adjustActiveItemPosition(SlideDirection.NEXT);
+        this._carouselService.pickNext(this.dir);
+        this._notifySlideChange(SlideDirection.NEXT);
+        this._changeDetectorRef.markForCheck();
+    }
+
+    /** @hidden Adjust position of active item, based on slide direction */
+    private _adjustActiveItemPosition(slideDirection: SlideDirection): void {
+        // Move one step in the direction
+        const positionAdjustment = slideDirection === SlideDirection.NEXT ? 1 : -1;
+        this.currentActiveSlidesStartIndex = this.currentActiveSlidesStartIndex + positionAdjustment;
+
+        // If carousel set to loop
+        if (this.loop) {
+            if (this.currentActiveSlidesStartIndex < 0) {
+                this.currentActiveSlidesStartIndex = this.slides.length - 1 - this.visibleSlidesCount + 1;
+            } else if (this.currentActiveSlidesStartIndex + this.visibleSlidesCount - 1 === this.slides.length) {
+                this.currentActiveSlidesStartIndex = 0;
+            }
+        } else {
+            this._buttonVisibility();
+        }
+    }
+
+    /** Handles navigation button visibility */
+    private _buttonVisibility(): void {
+        if (!this.loop) {
+            // Need to disable navigation button if either direction limit has reached.
+            if (this.currentActiveSlidesStartIndex === 0) {
+                this.leftButtonDisabled = true;
+            } else if (this.slides.length - 1 === this.currentActiveSlidesStartIndex) {
+                this.rightButtonDisabled = true;
+            } else if (
+                this.visibleSlidesCount > 1 &&
+                this.currentActiveSlidesStartIndex + this.visibleSlidesCount >= this.slides.length
+            ) {
+                this.rightButtonDisabled = true;
+            }
+
+            if (this.slides.length === 1) {
+                this.leftButtonDisabled = true;
+                this.rightButtonDisabled = true;
+            }
+            if (this.currentActiveSlidesStartIndex === 0) {
+                this.leftButtonDisabled = true;
+            } else if (this.currentActiveSlidesStartIndex === this.slides.length - 1) {
+                this.rightButtonDisabled = true;
+            }
+
+            if (this.currentActiveSlidesStartIndex > 0) {
```

Comment:
Could you please move `this.leftButtonDisabled = false;` and `this.rightButtonDisabled = false;` to above construction with `...else...`, and remove unnecessary conditions

`if (this.currentActiveSlidesStartIndex === 0) {`
`                this.leftButtonDisabled = true;`
`            } else if (this.currentActiveSlidesStartIndex === this.slides.length - 1) {`
`                this.rightButtonDisabled = true;`
`            } else {`
`this.leftButtonDisabled = false;`
`this.rightButtonDisabled = false;`
`}`

Reply:
fixed.